test/es-module/test-esm-loader-with-source.mjs

```diff
@@ -0,0 +1,10 @@
+// Flags: --experimental-loader ./test/fixtures/es-module-loaders/preset-cjs-source.mjs
+import '../common/index.mjs';
+import * as fixtures from '../common/fixtures.mjs';
+import assert from 'assert';
+
+const { default: existingFileSource } = await import(fixtures.fileURL('es-modules', 'cjs-file.cjs'));
+const { default: noSuchFileSource } = await import(fixtures.fileURL('no-such-file.cjs'));
```

Comment:
Thanks! Unfortunately, `'./no-such-file.cjs'` wouldn't work because it's not a valid URL, and the [`resolve` hook](https://github.com/nodejs/node/blob/c93377f61f8d95360e5b596d48ba7d4e290cda2e/test/fixtures/es-module-loaders/preset-cjs-source.mjs#L3-L7) doesn't actually resolve relative paths because of the short-circuit. Anyway, a file URL like `'file:///no-such-file.cjs'` works fine and is not relative (at least on Unix), so I switched to use that.

Reply:
I think it's incorrect to say it's not a valid URL, it is a valid (relative) URL.

> Anyway, a file URL like `'file:///no-such-file.cjs'` works fine

IMO it would make more sense to use a relative URL in the JS, and have the hook return a specific URL.

Reply:
Never mind, I had to change the test again because Windows doesn't like the absolute file URL without a drive letter: https://github.com/nodejs/node/pull/50465/commits/4dcbec57f99975322ca236b441e800647e5acfdf. Let's see how the CI build goes this time.